### PR TITLE
OSOE-795: Upgrade to latest OC preview to test System.Text.Json

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OSOE-795](https://lombiq.atlassian.net/browse/OSOE-795)

[OSOE-795]: https://lombiq.atlassian.net/browse/OSOE-795?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Lombiq.HelpfulExtensions/Extensions/CodeGeneration/CodeGenerationDisplayDriver.cs [hunk 1]
```diff
@@ -176,17 +167,17 @@ private string ConvertJArray(JArray jArray, int indentationDepth)
         return stringArrayCodeBuilder.ToString();
     }
 
-    private string ConvertJObject(JObject jObject, int indentationDepth)
+    private string ConvertJsonObject(JsonObject jObject, int indentationDepth)
```
Comment (dministro) [inline Lombiq.HelpfulExtensions/Extensions/CodeGeneration/CodeGenerationDisplayDriver.cs]:
```suggestion
    private string ConvertJsonObject(JsonObject jsonObject, int indentationDepth)
```

File: Lombiq.HelpfulExtensions/Extensions/ContentSets/Workflows/Handlers/WorkflowContentSetEventHandler.cs [hunk 2]
```diff
@@ -54,14 +54,16 @@ public class WorkflowContentSetEventHandler : IContentSetEventHandler
...
-                    links.AddRange(JToken.FromObject(objects).ToObject<IEnumerable<ContentSetLinkViewModel>>());
-                    break;
                 case ExpandoObject expandoObject:
-                    links.Add(JToken.FromObject(expandoObject).ToObject<ContentSetLinkViewModel>());
+                    links.Add(SerializeAndDeserialize<ContentSetLinkViewModel>(expandoObject));
+                    break;
+                case IEnumerable<object> collection when
+                    collection.CastWhere<ExpandoObject>().ToList() is { } objects &&
+                    objects.Count != 0:
+                    links.AddRange(SerializeAndDeserialize<IEnumerable<ContentSetLinkViewModel>>(objects));
```
Comment (dministro) [inline Lombiq.HelpfulExtensions/Extensions/ContentSets/Workflows/Handlers/WorkflowContentSetEventHandler.cs:63]: Why do we need this?

**Review by dministro — CHANGES_REQUESTED** [review]

File: Lombiq.HelpfulExtensions/Extensions/ContentSets/Workflows/Handlers/WorkflowContentSetEventHandler.cs (hunk 2 above)
Comment (sarahelsaig) [reply Lombiq.HelpfulExtensions/Extensions/ContentSets/Workflows/Handlers/WorkflowContentSetEventHandler.cs:63]: It was already there [a few lines above](https://github.com/Lombiq/Helpful-Extensions/pull/161/files/5800d7b01784a9d107f18207a563e3e4d18caf3c#diff-cde8ef6bc1c3fa66dfdfb10a159625b6808706184a53db3cbd3ad9982bd37a84L58).

Comment (dministro) [reply Lombiq.HelpfulExtensions/Extensions/ContentSets/Workflows/Handlers/WorkflowContentSetEventHandler.cs:63]: I see. You are right.

**Review by dministro — APPROVED** [review]

**Pull request merged**